src/vs/vscode.d.ts

```diff
@@ -4967,6 +4967,11 @@ declare module 'vscode' {
 		 */
 		label: string;
 
+		/**
+		 * Optional id for the tree item.
```

Comment:
This needs *a lot more* documentation which explains what the id is and what it is used for, why it optional etc. I'd say much of this: https://github.com/Microsoft/vscode/issues/40018#issuecomment-356890904

Reply:
Added more doc to the `id` attribute explaining why it is used and the default behaviour.